File: app/src/main/java/com/android/ashwiask/tvmaze/AppModule.java
package com.android.ashwiask.tvmaze;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ashwini Kumar.
 */
@Module
public class AppModule
{
    private TvMazeApplication application;

    public AppModule(TvMazeApplication app)
    {
        application = app;
    }

    @Provides
    @Singleton
    Context provideContext()
    {
        return application;
    }

    @Provides
    Application provideApplication()
    {
        return application;
    }
}

File: app/src/main/java/com/android/tvmaze/di/AppModule.kt
package com.android.tvmaze.di

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import javax.inject.Singleton

@InstallIn(ApplicationComponent::class)
@Module
object AppModule {
    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application
    }
}
